main.c

```diff
@@ -1,189 +1,199 @@
+    {
+        mtx_p->start[i] = (double*)malloc(mtx_p->M*sizeof(double));
+        if (mtx_p->start[i] == NULL) {
+            printf("Error: Memory allocation failed\n");
+            return 1;
+        }
+    }
+    return 0;
+}
+
+int load_matrix(char* file_name, struct matrix* mtx_p) {
+    FILE* fp;
+    char * line = NULL;
+    size_t len = 0;
+    ssize_t read;
+
+    int rows_counter = 0;
+
+    fp = fopen(file_name, "r");
+    if (fp == NULL) {
+        printf("Error: File %s cannot be open\n", file_name);
+        return 1;
+    }
+
+    // Allocate space for the matrix
+    int ret = 0;
+    ret = allocate_matrix(mtx_p);
+    if (ret) {
+        return 1; // malloc failed
+    }
+    
+    // https://stackoverflow.com/questions/3501338/c-read-file-line-by-line
+    int EOF_found = 0;
+    for (int i = 0; i < mtx_p->N && !EOF_found; i++) {
+        for (int j = 0; j < mtx_p->M && !EOF_found; j++) {
+            read = getline(&line, &len, fp);
+            if (read == EOF) {
+                EOF_found = 1;
+            }
+
+            // Check if file has more numbers than expected
+            if (rows_counter > mtx_p->size) {
+                printf("Error: Matrix is larger than specified\n");
+                free_matrix(mtx_p);
+                return 1;
+            }
+
+            mtx_p->start[i][j] = atof(line);
+            rows_counter++;
+        }
+    }
+    
+    // Check if file has less numbers than expected
+    if (rows_counter < mtx_p->size) {
+        printf("Error: Matrix is smaller than specified\n");
+        free_matrix(mtx_p);
+        return 1;
+    }
+
+    fclose(fp);
+    if (line) {
+        free(line);
+    }
+    return 0;
+}
+
+void read_matrix_size(struct matrix* mtx_p) {
+    char input_buffer[MAX_INPUT_LEN];
+
+    printf("Insert number of rows N: ");
+    fgets(input_buffer, MAX_INPUT_LEN, stdin);
+    mtx_p->N = atoi(input_buffer);
+    //printf("N: %d\n", mtx->N);
+
+    printf("Insert number of columns M: ");
+    fgets(input_buffer, MAX_INPUT_LEN, stdin);
+    mtx_p->M = atoi(input_buffer);
+    //printf("M: %d\n", mtx->M);
+
+    mtx_p->size = mtx_p->N * mtx_p->M; // Compute total number of elements
+}
+
+int main(int argc, char const *argv[])
+{
+    struct matrix matrix_A, matrix_B, matrix_C;
+
+    // Ask for matrix A size
+    printf("Matrix A\n");
+    read_matrix_size(&matrix_A);
+    
+    // Ask for matrix B size
+    printf("Matrix B\n");
+    read_matrix_size(&matrix_B);
+
+    // Check for valid multiplication
+    if (matrix_A.M != matrix_B.N) {
+        printf("Error: Matrices cannot be multiplied!\n");
+        return 1;
+    }
+
+    // Load matrices
+    int ret;
+    ret = load_matrix("test/matrixA.txt", &matrix_A);
+    ret = load_matrix("test/matrixB.txt", &matrix_B);
+    if (ret) {
+        printf("Error: Matrices were not loaded correctly\n");
+    }
+
+    double temp = 0;
+    //Transpose matrix_B
+    for (int r=0; r < matrix_B.rows; r++){

```

Comment:
Could you make this its own function? Something like `void transpose_matrix(struct matrix* mtx)`. That could allow us to reuse the function later on.